test/unit/Types/ScalarTypesResolverTest.php

```diff
@@ -0,0 +1,210 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OnMoon\OpenApiServerBundle\Test\Unit\Types;
+
+use cebe\openapi\spec\Type;
+use OnMoon\OpenApiServerBundle\Types\ScalarTypesResolver;
+use PHPUnit\Framework\Assert;
+use PHPUnit\Framework\TestCase;
+use Safe\DateTime;
+use Safe\Exceptions\DatetimeException;
+
+/**
+ * @covers  \OnMoon\OpenApiServerBundle\Types\ScalarTypesResolver
+ */
+class ScalarTypesResolverTest extends TestCase
+{
+    private ScalarTypesResolver $scalarTypeResolver;
+
+    protected function setUp(): void
+    {
+        $this->scalarTypeResolver = new ScalarTypesResolver();
+    }
+
+    /**
+     * @return array|mixed[]
+     *
+     * @throws DatetimeException
+     */
+    public function convertDeserializeDataProvider(): array
+    {
+        return [
+            'deserialize_null' => [
+                'deserialize' => true,
+                'id' => 1,
+                'value' => null,
+                'expectedValue' => null,
+            ],
+            'deserialize_date' => [
+                'deserialize' => true,
+                'id' => 1,
+                'value' => '2020-07-31',
+                'expectedValue' => DateTime::createFromFormat('Y-m-d', '2020-07-31'),
+            ],
+            'deserialize_date_time' => [
+                'deserialize' => true,
+                'id' => 2,
+                'value' => '2020-07-31 15:35:11',
+                'expectedValue' => new DateTime('2020-07-31 15:35:11'),
+            ],
+            'deserialize_byte' => [
+                'deserialize' => true,
+                'id' => 3,
+                'value' => 'VGhpcyBpcyBhbiBlbmNvZGVkIHN0cmluZw==',
+                'expectedValue' => 'This is an encoded string',
+            ],
+            'deserialize_php_type_bool' => [
+                'deserialize' => true,
+                'id' => 11,
+                'value' => '',
+                'expectedValue' => false,
+            ],
+            'deserialize_php_type_float' => [
+                'deserialize' => true,
+                'id' => 5,
+                'value' => '1.1qwerty',
```

Comment:
The basic case here is to test for a string containing a correct float, no a string that can possibly be type-juggled to some float value.